src/blockedtuple.jl

```diff
@@ -0,0 +1,76 @@
+# This file defines BlockedTuple, a Tuple of heterogeneous Tuple with a BlockArrays.jl
+# like interface
+
+using BlockArrays: Block, BlockArrays, BlockIndexRange, BlockRange, blockedrange
+
+struct BlockedTuple{Divs,Flat}
+  flat::Flat
+
+  function BlockedTuple{Divs}(flat::Tuple) where {Divs}
+    length(flat) != sum(Divs) && throw(DimensionMismatch("Invalid total length"))
+    return new{Divs,typeof(flat)}(flat)
+  end
+end
+
+# TensorAlgebra Interface
+BlockedTuple(tt::Vararg{Tuple}) = BlockedTuple{length.(tt)}(flatten_tuples(tt))
```

Comment:
I think we should follow the BlockArrays.jl constructor conventions as closely as possible. That would mean the constructors would be:
```julia
tuplemortar(((1, 2), (3, 4, 5))) == BlockedTuple{(2, 3)}((1, 2, 3, 4, 5))
BlockedTuple((1, 2, 3, 4, 5), (2, 3)) == BlockedTuple{(2, 3)}((1, 2, 3, 4, 5))
```